library/src/main/java/com/novoda/downloadmanager/NetworkFileDownloader.java

```diff
@@ -60,6 +46,28 @@ public void startDownloading(String url, FileSize fileSize, Callback callback) {
         callback.onDownloadFinished();
     }
 
+    private void processResponse(Callback callback, HttpClient.NetworkResponse response, int responseCode) throws IOException {
```

Comment:
I wasn't sure how to break this code down more but it is hitting the cyclomatic complexity limit 😬